_posts/bulk-editing-garmin-data.md

```diff
@@ -0,0 +1,129 @@
+
+We can achieve the desired outcome by using the following code:
+
+```javascript
+jQuery.getJSON(
+    '/modern/proxy/activitylist-service/activities/{gear-id}/gear?start=1&limit=25',
+    function(gear_activities){
+        var virtualRides = gear_activities.filter(function(act){
+                return act.activityType.typeKey == "virtual_ride";
+            });
+
+        if(virtualRides.length == 0){
+            return;
+        }
+
+        virtualRides.forEach(function(act){
+            //unlink
+            fetch('/modern/proxy/gear-service/gear/unlink/{gear-id}/activity/' + act['activityId'],
+            {
+                'headers': { 'Content-Type': 'application/json', 'nk': 'NT', 'X-HTTP-Method-Override': 'PUT' },
+                'method': 'POST'
+            });    
+
+            //link
+            fetch('/modern/proxy/gear-service/gear/link/{gear-id}/activity/' + act['activityId'],
+            {
+                'headers': { 'Content-Type': 'application/json', 'nk': 'NT', 'X-HTTP-Method-Override': 'PUT' },
+                'method': 'POST'
+            });
+        });
+    });
+```
+
+It's not that complex. We first retrieve a few activities:
+
+```javascript
+jQuery.getJSON(
+    '/modern/proxy/activitylist-service/activities/{gear-id}/gear?start=1&limit=25',
+```
+
+The above snippet retrieves the top 25 activities for a given gear.
+
+> Your mileage may vary with the 25 items limit. I run the above script more or less every couple of weeks. The query returns activities ordered by date, with the latest on top. I assume that there is no way I can record more than 25 activities for a specific gear in two weeks.
+
+The next step is to filter to get only virtual activities:
+
+```javascript
+var virtualRides = gear_activities.filter(function(act){
+   return act.activityType.typeKey == "virtual_ride";
+});
+
+if(virtualRides.length == 0){
+   return;
+}
+```
+
+If there are no virtual activities, we end prematurely. Finally, the script iterates over all the virtual activities, removes the incorrectly assigned gear, and assigns the desired one:
+
+```javascript
+virtualRides.forEach(function(act){
+   //unlink
+   fetch('/modern/proxy/gear-service/gear/unlink/{gear-id}/activity/' + act['activityId'],
+   {
+       'headers': { 'Content-Type': 'application/json', 'nk': 'NT', 'X-HTTP-Method-Override': 'PUT' },
+       'method': 'POST'
+    });    
+
+     //link
+     fetch('/modern/proxy/gear-service/gear/link/{gear-id}/activity/' + act['activityId'],
+     {
+        'headers': { 'Content-Type': 'application/json', 'nk': 'NT', 'X-HTTP-Method-Override': 'PUT' },
+        'method': 'POST'
+});
+```
+
+> Disclaimer: It's not my work. I limited myself to putting together snippets found on Garmin forums.
+
+## Conclusion
+
+First, if you are a Garmin user and need to bulk edit data, the presented solution might work out of the box or be adjusted to your needs.

```

Comment:
```suggestion
First, if you are a Garmin user and need to bulk edit data, the presented solution might work out of the box or it might need to be adjusted to your needs.

```